browser/net/brave_static_redirect_network_delegate_helper.cc

```diff
@@ -8,13 +8,35 @@
 #include <memory>
 #include <vector>
 
+#include "base/strings/stringprintf.h"
 #include "brave/browser/translate/buildflags/buildflags.h"
 #include "brave/common/network_constants.h"
 #include "brave/common/translate_network_constants.h"
 #include "extensions/common/url_pattern.h"
 
+#if !defined(BRAVE_SERVICES_KEY)
+#define BRAVE_SERVICES_KEY "dummytoken"
+#endif
+
+const char kBraveServicesKeyHeader[] = "BraveServiceKey";
+
 namespace brave {
 
+int OnBeforeStartTransaction_StaticRedirectWork(
+    net::HttpRequestHeaders* headers,
+    const ResponseCallback& next_callback,
+    std::shared_ptr<BraveRequestInfo> ctx) {
+
+  static URLPattern proxy_pattern(URLPattern::SCHEME_HTTP |
+      URLPattern::SCHEME_HTTPS, kBraveProxyPattern);
+
+  if (proxy_pattern.MatchesURL(ctx->request_url)) {
```

Comment:
I'm not sure this is a viable method to use anymore with network service - cc @iefremov

Reply:
I'm also not sure this was ever really the right place to add headers

Reply:
why not? I tested `OnBeforeStartTransaction_SiteHacksWork` and it worked for UA overriding. It should be safe for non-restricted headers (Cookies, Referer, etc)

Reply:
It's definitely not the right place for UA overriding because there are methods specifically for that. The other method I was thinking of was NavigationHandle::SetRequestHeaders but that only works for main/subframe navigations